iati/tests/test_resources.py

```diff
@@ -1,32 +1,151 @@
+        file_name_regex = re.compile(r'^([a-z]+[\-_])*[a-z]+$')
 
-class TestResourceFolders(object):
-    """A container for tests relating to resource folders."""
+        assert re.match(file_name_regex, file_name)
+
+    @pytest.mark.parametrize('file_extension', [
+        iati.resources.FILE_CODELIST_EXTENSION,
+        iati.resources.FILE_DATA_EXTENSION,
+        iati.resources.FILE_RULESET_EXTENSION,
+        iati.resources.FILE_SCHEMA_EXTENSION
+    ])
+    def test_file_extensions_valid_values(self, file_extension):
+        """Test that constants that should be file extensions are a dot followed by a lower case string."""
+        file_extension_regex = re.compile(r'^\.[a-z]+$')
+
+        assert re.match(file_extension_regex, file_extension)
+
+
+@pytest.mark.new_tests
+class TestResourceFilesystemPaths(object):
+    """A container for tests relating to specific filesystem paths."""
+
+    def test_resource_filesystem_path(self, filename_no_meaning):
+        """Check that resource file names are found correctly."""
+        base_path = iati.resources.resource_filesystem_path('')
+        full_path = iati.resources.resource_filesystem_path(filename_no_meaning)
+
+        assert len(full_path) > len(filename_no_meaning)
+        assert full_path.startswith(base_path)
+        assert full_path.endswith(filename_no_meaning)
+        assert os.path.abspath(full_path) == full_path
+
+    def test_resource_filesystem_path_folders(self, folderpath_no_meaning):
+        """Check that resource folder names are found correctly."""
+        base_path = iati.resources.resource_filesystem_path('')
+        full_path = iati.resources.resource_filesystem_path(folderpath_no_meaning)
+
+        assert len(full_path) > len(folderpath_no_meaning)
+        assert full_path.startswith(base_path)
+        assert full_path.endswith(folderpath_no_meaning)
+        assert os.path.abspath(full_path) + os.path.sep == full_path
+
+    def test_resource_filesystem_path_empty_path(self, filepath_empty):
+        """Check that the base resource folder is located when given an empty filepath."""
+        full_path = iati.resources.resource_filesystem_path(filepath_empty)
+
+        assert full_path != ''
+        assert os.path.isdir(full_path)
+
+
+@pytest.mark.new_tests
+class TestResourceLibData(object):
+    """A container for tests relating to handling paths for pyIATI library-specific data."""
+
+    def test_create_lib_data_path(self, filename_no_meaning):
+        """Check that library data can be located."""
+        full_path = iati.resources.create_lib_data_path(filename_no_meaning)
+
+        assert iati.resources.BASE_PATH_LIB_DATA in full_path
+        assert full_path.endswith(filename_no_meaning)
+
+
+@pytest.mark.new_tests
+class TestResourceHandlingInvalidPaths(object):
+    """A container for tests relating to handling paths that are invalid and being passed to functions that are version-independent."""
```

Comment:
This docstring should be improved to better clarify why this class is its own thing.